### DISCUSSION TITLE
Seeking Assistance with Electrode Solver

### DISCUSSION THREAD

**Discussion by kimo707** [pybamm-team/PyBaMM#3966] [Q&A] [answered]
Hey,

I'm currently working on a piece of code where I need to update the volume fraction to reflect changes in the active material volume fractions and the amount of cyclable lithium. This is crucial for capturing the primary effects of degradation in later cycles. To accomplish this, I'm trying to use an ElectrodeSOHSolver to calculate the stoichiometries 

​The challenge I'm facing is the need to start each simulation from the endpoint of the previous one, continuing from that specific state to recalibrate the stoichiometries. I attempted to utilize the statement esoh_solver.set_initial_conditions_from(esoh_sol) to preserve the last solution and restart from that juncture, but it hasn’t worked.

Here's a snippet of my code for context:

        inputs = {"V_min": Vmin, "V_max": Vmax, "Q_n": Q_n, "Q_p": Q_p, "Q_Li": n_Li * pybamm.constants.F.value/3600 }
        esoh_solver = pybamm.lithium_ion.ElectrodeSOHSolver(parameter_values, param)
        
        esoh_sol = esoh_solver.solve(inputs)
        ##esoh_solver.built_model.set_initial_conditions_from(esoh_sol)
        ##esoh_solver.set_initial_conditions_from(esoh_sol)
        
        x_100 = esoh_sol["x_100"]
        
        y_100 = esoh_sol["y_100"]
        x_0 = esoh_sol["x_0"]
        y_0 = esoh_sol["y_0"]

Could you provide any advice or insights on how to resolve this issue?

**Comment by valentinsulzer** [comment, marked as answer]
>  need to start each simulation from the endpoint of the previous one, continuing from that specific state to recalibrate the stoichiometries

You don't need to do that. Just do 
```python
Q_n = solution["Negative electrode capacity [A.h]"].data[-1]
Q_p = solution["Positive electrode capacity [A.h]"].data[-1]
Q_Li = solution["Total lithium capacity in particles [A.h]"].data[-1]
inputs = {"Q_n": Q_n, "Q_p": Q_p, "Q_Li": Q_Li}
```
`Q_Li` captures the change in stoichiometries

**Reply by kimo707** [reply]
Thank you
